neo/io/blackrockio.py

```diff
@@ -538,8 +538,9 @@ def __read_nsx_dataheader_variant_b(
                 'offset_to_data_block': offset + dh.dtype.itemsize}
 
             # data size = number of data points * (2bytes * number of channels)
-            data_size = dh['nb_data_points'] * \
-                self.__nsx_basic_header[nsx_nb]['channel_count'] * 2
+            #int for avoid overflow problem
+            data_size = int(dh['nb_data_points']) * \
+                int(self.__nsx_basic_header[nsx_nb]['channel_count']) * 2
             # define new offset (to possible next data block)
```

Comment:
Yes, we didn't realize this could be problematic with large file sizes, good point.